packages/tangle/parser.go

```diff
@@ -248,6 +251,157 @@ type MessageFilter interface {
 	OnReject(callback func(msg *Message, err error, peer *peer.Peer))
 }
 
+/**
+Syntactical checks for a message
+*/
+type MessageSyntacticalValidator struct {
+	onAcceptCallback func(msg *Message, peer *peer.Peer)
+	onRejectCallback func(msg *Message, err error, peer *peer.Peer)
+
+	onAcceptCallbackMutex sync.RWMutex
+	onRejectCallbackMutex sync.RWMutex
+}
+
+func NewMessageSyntacticalValidator() *MessageSyntacticalValidator {
+	return &MessageSyntacticalValidator{}
+}
+
+/**
+Performs ths following syntactical checks:
+1. A Strong Parents Block must exist.
+2. Parents Block types cannot repeat.
+3. Parent count per block 0 <= x <= 8.
+4. Parents unique within block.
+5. Parents lexicographically sorted within block.
+6. ParentsBlockCount must match blocks actually parsed.
+7. ParentsCount within each block must match the parents actually parsed within the block.
+8. A Parent(s) repetition is only allowed when it occurs across Strong and Like parents
+8. Blocks should be ordered by type in ascending order
+**/
+func (f *MessageSyntacticalValidator) Filter(msg *Message, peer *peer.Peer) {
+	// A strong Block must exist
+	if len(msg.ParentsByType(StrongParentType)) == 0 {
+		f.onRejectCallback(msg, ErrNoStrongParents, peer)
+		return
+	}
+
+	// validate block count
+	if int(msg.parentsBlocksCount) != len(msg.parentsBlocks) {
```

Comment:
Hmm it only holds it implicitely (you need to get the length of the slice)

The reason it is there as far as I undestand is because `parentsBlockCount` is part of the message layout and how it is serialized, and we want the struct to reflect that


I guess once we do validation via the serializer we can give up on this field and use only the slice

Reply:
looking at the code again, I actually agree that it can be removed now already

Reply:
I said I will remove it but:
1. There are too many changes in this PR
2. The current comment in the code that says that the struct should reflect what is sent over the wire made me feel a bit uneasy

Reply:
Tracked here: https://github.com/iotaledger/goshimmer/issues/1607, can be done at a later point in time.